pkg/daemon/daemon.go

```diff
@@ -1334,34 +1446,8 @@ func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) error
 	if !osMatch {
 		return errors.Errorf("expected target osImageURL %q, have %q", currentConfig.Spec.OSImageURL, dn.bootedOSImageURL)
 	}
-	// And the rest of the disk state
-	// We want to verify the disk state in the spec version that it was created with,
-	// to remove possibilities of behaviour changes due to translation
-	ignconfigi, err := ctrlcommon.IgnParseWrapper(currentConfig.Spec.Config.Raw)
-	if err != nil {
-		return errors.Errorf("Failed to parse Ignition for validation: %s", err)
-	}
 
-	switch typedConfig := ignconfigi.(type) {
-	case ign3types.Config:
-		if err := checkV3Files(ignconfigi.(ign3types.Config).Storage.Files); err != nil {
-			return err
-		}
-		if err := checkV3Units(ignconfigi.(ign3types.Config).Systemd.Units); err != nil {
-			return err
-		}
-		return nil
-	case ign2types.Config:
-		if err := checkV2Files(ignconfigi.(ign2types.Config).Storage.Files); err != nil {
-			return err
-		}
-		if err := checkV2Units(ignconfigi.(ign2types.Config).Systemd.Units); err != nil {
-			return err
-		}
-		return nil
-	default:
-		return errors.Errorf("unexpected type for ignition config: %v", typedConfig)
-	}
+	return validateOnDiskState(currentConfig, pathSystemd)
```

Comment:
Instead of passing const value pathSystemd to all the places. Maybe move this const variable to https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/constants/constants.go

Reply:
Yes. For easy cleanup, the Config Drift Monitor test suite creates a tempdir to write the Ignition files, systemd unit and dropin files to. To detect the systemd unit and dropin files, I modified `validateOnDiskState` (and some of its helpers) to accept a `systemdPath` argument instead of having `pathSystemd` hard-coded. That said, Config Drift Monitor, as well as `validateOnDiskState's` helpers will default to `pathSystemd` if an empty string is passed in.